packages/conventional-recommended-bump/index.js

```diff
@@ -4,86 +4,88 @@ var conventionalCommitsFilter = require('conventional-commits-filter');
 var conventionalCommitsParser = require('conventional-commits-parser');
 var gitSemverTags = require('git-semver-tags');
 var gitRawCommits = require('git-raw-commits');
-var objectAssign = require('object-assign');
+var Q = require('q');
 
 var VERSIONS = ['major', 'minor', 'patch'];
 
-function conventionalRecommendedBump(options, parserOpts, cb) {
-  var config;
-  var noop = function() {};
+module.exports = conventionalRecommendedBump;
 
+function conventionalRecommendedBump(options, parserOpts, cb) {
   if (typeof options !== 'object') {
     throw new TypeError('options must be an object');
   }
 
-  if (typeof parserOpts === 'function') {
-    cb = parserOpts;
-  } else {
-    cb = cb || noop;
-  }
+  options = Object.assign({ignoreReverted: true, warn: function() {}}, options);
 
-  options = objectAssign({
-    ignoreReverted: true,
-    warn: function() {}
-  }, options);
+  cb = typeof parserOpts === 'function' ? cb = parserOpts : cb = cb || noop;
 
+  let presetPackage;
   if (options.preset) {
     try {
-      config = require('./presets/' + options.preset);
+      presetPackage = require(`conventional-changelog-${options.preset.toLowerCase()}`);
```

Comment:
What do you think of removing preset option and adding a `config.extends` (similar to what eslint does). It accepts a string (or an array of strings) of either relative or absolute path to a module.

Reply:
This will prevent scope packages from working

Reply:
@fenduru this would simplify scope packages, since a path is more generic.

Reply:
> What do you think of removing preset option and adding a config.extends (similar to what eslint does). It accepts a string (or an array of strings) of either relative or absolute path to a module.

@stevemao we could do that. Though that would mean `conventional-changelog` and `conventional-recommended-bump` would remain different in how they load presets (unless we rolled `conventional-changelog` into this PR.)

Reply:
> This will prevent scope packages from working

My current implementation does not support npm scopes (`@my-org`). That is correct.

My hope is to borrow the implementation from #272, once complete.

> adding a config.extends

@stevemao if you're interested in that kind of change to the API, maybe we should roll it into #272 where we're changing the behavior of presets to add scope support.

It should be noted that ESLint [behaves](https://eslint.org/docs/developer-guide/shareable-configs#using-a-shareable-config) similar to the work in #272 _plus_ support for arbitrary paths, though ESLint requires that you specify the full package name when using scopes, instead of assuming the package name like Yeoman, Yarn, and #272 do.